src/QsCompiler/Tests.Compiler/TestCases/TypeParameterResolution.qs

```diff
@@ -157,3 +157,110 @@ namespace Microsoft.Quantum.Testing.TypeParameterResolution {
 
 // Foo.A -> Int
 // Bar.B -> Int
+
+// =================================
```

Comment:
As a general note on systematic approach to test case generation:
1. Identify interesting callgraph topology equivalence classes without entry point attribute being involved and create corresponding test cases for each.
2. Consider each of these cases and think whether adding the attribute somewhere _inside_ the callgraph creates a new test case. E.g. right now you are not covering the case when there is a caller of the method with entry point attribute. If the compiler supports user code like this, should this caller of the entry point be included into the tree or not?
3. Hopefully, multiple entry point attributes aren't allowed, but if they were, would need to test that as well.

Reply:
For 3: I can't currently write tests that have more that one entry point in them as that is disallowed. It doesn't seem like we are going to allow multiple entry points in the future, but if we do then we will address the changes that will entail then.

For 2: That is a good idea for some tests. I will look into adding such tests.

Reply:
I've added some more tests related to when entry points are called from other operations.